components/core/src/clp/ffi/ir_stream/utils.hpp

```diff
@@ -43,5 +66,23 @@ auto serialize_int(integer_t value, std::vector<int8_t>& ir_buf) -> void {
     std::span<int8_t> const data_view{reinterpret_cast<int8_t*>(&value_big_endian), sizeof(value)};
     ir_buf.insert(ir_buf.end(), data_view.begin(), data_view.end());
 }
+
+template <typename encoded_variable_t>
+[[nodiscard]] auto serialize_clp_string(std::string_view str, std::vector<int8_t>& buf) -> bool {
+    static_assert(
+            (std::is_same_v<encoded_variable_t, clp::ir::eight_byte_encoded_variable_t>
+             || std::is_same_v<encoded_variable_t, clp::ir::four_byte_encoded_variable_t>)
+    );
+    std::string logtype;
```

Comment:
nitpick: could we re-use the same buffer for clp logtype instead of recreating it each time this function is called?